garrysmod/gamemodes/terrortown/gamemode/gamemsg.lua

```diff
@@ -91,54 +92,59 @@ end
+	   return true 
+	end
+	
+	return false
+end
+
+local mumbles = {"mumble", "mm", "hmm", "hum", "mum", "mbm", "mble", "ham", "mammaries", "political situation", "mrmm", "hrm", 
+"uzbekistan", "mumu", "cheese export", "hmhm", "mmh", "mumble", "mphrrt", "mrh", "hmm", "mumble", "mbmm", "hmml", "mfrrm"}
 
 -- While a round is active, spectators can only talk among themselves. When they
 -- try to speak to all players they could divulge information about who killed
 -- them. So we mumblify them. In detective mode, we shut them up entirely.
 function GM:PlayerSay(ply, text, team_only)
-   if not IsValid(ply) then return end
-
-   if team_only and ply:Team() != TEAM_SPEC and GetRoundState() == ROUND_ACTIVE then
-      if ply:IsSpecial() then
-         -- traitor chat handling
-         RoleChatMsg(ply, ply:GetRole(), text)
-      else
-         LANG.Msg(ply, "inno_globalchat_hint")
-      end
-
-      return ""
-   end
-
-   if (not team_only) and GetRoundState() == ROUND_ACTIVE and ply:Team() == TEAM_SPEC then
-      if DetectiveMode() then
-         LANG.Msg(ply, "spec_teamchat_hint")
-         return ""
-      end
-
-      if not GetConVar("ttt_limit_spectator_chat"):GetBool() then
-         return text
-      end
+   if not IsValid(ply) then return "" end
+	
+   if GetRoundState() == ROUND_ACTIVE then
+      local team = ply:Team() == TEAM_SPEC
+      if team and not DetectiveMode() then
+         local filtered = {}
+         for k, v in pairs(string.Explode(" ", text)) do
+            -- grab word characters and whitelisted interpunction
+            -- necessary or leetspeek will be used (by trolls especially)
+            local word, interp = string.match(v, "(%a*)([%.,;!%?]*)")
+            if word != "" then
+               table.insert(filtered, mumbles[math.random(1, #mumbles)] .. interp)
+            end
+         end
 
-      local filtered = {}
-      for k, v in pairs(string.Explode(" ", text)) do
-         -- grab word characters and whitelisted interpunction
-         -- necessary or leetspeek will be used (by trolls especially)
-         local word, interp = string.match(v, "(%a*)([%.,;!%?]*)")
-         if word != "" then
-            table.insert(filtered, mumbles[math.random(1, #mumbles)] .. interp)
+         -- make sure we have something to say
+         if table.Count(filtered) < 1 then
+            table.insert(filtered, mumbles[math.random(1, #mumbles)])
          end
-      end
 
-      -- make sure we have something to say
-      if table.Count(filtered) < 1 then
-         table.insert(filtered, mumbles[math.random(1, #mumbles)])
+         table.insert(filtered, 1, "[MUMBLED]")
+         return table.concat(filtered, " ")
+      elseif team_only and not team and ply:IsSpecial() then
+	     RoleChatMsg(ply, ply:GetRole(), text)
+		 return ""
       end
-
-      table.insert( filtered, 1, "[MUMBLED]")
-      return table.concat(filtered, " ")
    end
-
+   

```

Comment:
Submit your own PR; I'm not adding another commit, pal :last_quarter_moon_with_face: